src/ProgressMeter.jl

```diff
@@ -177,67 +192,66 @@ macro showprogress(args...)
     metersym = gensym("meter")
 
     if isa(loop, Expr) && loop.head === :for
-        @assert length(loop.args) == 2
-        assignidx = 1
-        loopbodyidx = 2
-        is_comprehension = false
+        outerassignidx = 1
+        loopbodyidx = endof(loop.args)
     elseif isa(loop, Expr) && loop.head in (:comprehension, :dict_comprehension)
-        @assert length(loop.args) == 2
-        assignidx = 2
+        outerassignidx = endof(loop.args)
         loopbodyidx = 1
-        is_comprehension = true
     elseif isa(loop, Expr) && loop.head in (:typed_comprehension, :typed_dict_comprehension)
-        @assert length(loop.args) == 3
-        assignidx = 3
+        outerassignidx = endof(loop.args)
         loopbodyidx = 2
-        is_comprehension = true
     else
         throw(ArgumentError("Final argument to @showprogress must be a for loop or comprehension."))
     end
 
-    newloop = Expr(loop.head, loop.args...)
+    loop = copy(loop)
```

Comment:
I assume it's OK since the typical usage is for them to come from parsing a text file. But to clarify, I don't have strong feelings one way or another.